### DISCUSSION TITLE
Additional information for actions

### DISCUSSION THREAD

**Discussion by rkaldung** [tobimori/kirby-dreamform#76] [General] [answered]
I'd like to pass on additional information to the actions when the form is submitted. I use the visitor's IP address to determine the country of origin. Is there any way to "inject" such info?

**Comment by tobimori** [comment, marked as answer]
You can configure DreamForm to collect the IP address with the following option

```php
// site/config/config.php

return [
  'tobimori.dreamform' => [
    'metadata' => [
        'collect' => [
            'ip', 
			// [...]
        ]
    ],
  ],
];
```

and then access the IP like so:

`$submission->metadata()->ip()`

or in the Email action, you should be able to use

`{{ submission.metadata.ip }}`

For other actions, like the Mailchimp action, the IP address is automatically included when the metadata collect option is enabled.
